pkg/lib/cockpit.js

```diff
@@ -316,7 +316,8 @@ function calculate_url(suffix) {
     if (!suffix)
         suffix = "socket";
     const window_loc = window.location.toString();
-    let _url_root = url_root;
+    const meta_websocket_root = document.head.querySelector("meta[name='websocket-root']");
```

Comment:
Summary from IRC and meeting: we're gonna split these two lines into a separate commit which we merge today.  The rest of it will get compiled into our hacked up version of `cockpit-ws` that we used when testing the cloud side of this all.

Once we're sure that the entire solution is viable, we'll merge the rest.  If not, we can trivially back out these two lines, without any compatibility concerns.

Reply:
Done. Review is fine and appreciated, but please don't merge this yet before I re-run my PoCs against the COPR of *this* PR. I want to b 100% sure that I wasn't missing anything that was necessary on the client side.

Reply:
I did that testing now, and it's all :+1: Unblocked.